File: firestorelivedata/src/test/java/com/ptrbrynt/firestorelivedata/TaskResultTest.java
package com.ptrbrynt.firestorelivedata;

import org.junit.Test;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertTrue;

public class TaskResultTest {

    @Test
    public void success() {
        TaskResult<String> result = TaskResult.Companion.success("hello");
        assertEquals(result.getStatus(), TaskStatus.SUCCESS);
        assertEquals(result.getData(), "hello");
        assertEquals(result.getException(), null);
    }

    @Test
    public void failure() {
        TaskResult<String> result = TaskResult.Companion.failure(new IllegalStateException());
        assertEquals(result.getStatus(), TaskStatus.FAILED);
        assertEquals(result.getData(), null);
        assertTrue(result.getException() instanceof IllegalStateException);
    }

    @Test
    public void cancelled() {
        TaskResult<String> result = TaskResult.Companion.cancelled();
        assertEquals(result.getStatus(), TaskStatus.CANCELLED);
        assertEquals(result.getData(), null);
        assertEquals(result.getException(), null);
    }

    @Test
    public void running() {
        TaskResult<String> result = TaskResult.Companion.running();
        assertEquals(result.getStatus(), TaskStatus.RUNNING);
        assertEquals(result.getData(), null);
        assertEquals(result.getException(), null);
    }

}

File: firestorelivedata/src/test/java/com/ptrbrynt/firestorelivedata/TaskResultTest.kt
package com.ptrbrynt.firestorelivedata

import org.junit.Test

import org.junit.Assert.assertEquals
import org.junit.Assert.assertTrue

/**
 * Tests the [TaskResult] class
 */
class TaskResultTest {

    /** Tests the success state **/
    @Test
    fun success() {
        val result = TaskResult.success("hello")
        assertEquals(result.status, TaskStatus.SUCCESS)
        assertEquals(result.data, "hello")
        assertEquals(result.exception, null)
    }

    /** Tests the failure state **/
    @Test
    fun failure() {
        val result = TaskResult.failure<String>(IllegalStateException())
        assertEquals(result.status, TaskStatus.FAILED)
        assertEquals(result.data, null)
        assertTrue(result.exception is IllegalStateException)
    }

    /** Tests the cancelled state **/
    @Test
    fun cancelled() {
        val result = TaskResult.cancelled<String>()
        assertEquals(result.status, TaskStatus.CANCELLED)
        assertEquals(result.data, null)
        assertEquals(result.exception, null)
    }

    /** Tests the running state **/
    @Test
    fun running() {
        val result = TaskResult.running<String>()
        assertEquals(result.status, TaskStatus.RUNNING)
        assertEquals(result.data, null)
        assertEquals(result.exception, null)
    }

}
